tee-worker/core/parentchain/indirect-calls-executor/src/lib.rs

```diff
@@ -238,7 +238,7 @@ where
 		&self,
 		block_hash: H256,
 		extrinsics: Vec<H256>,
-		block_number: <<ParentchainBlock as ParentchainBlockTrait>::Header as Header>::Number,
+		block_number: u32,
```

Comment:
I think this trait implements `TryFrom<u32>` have you tried doing something like 
```rust
let block: u32 = block_number.try_into(); 
```